analyzers/tests/SonarAnalyzer.CSharp.Styling.Test/TestCases/UseRegexSafeIsMatch.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Text.RegularExpressions;
+using RealRegex = System.Text.RegularExpressions.Regex;
+
+class UseRegexSafeIsMatchNonCompliant
+{
+    private Regex regex;
+    private RealRegex realRegex;
+
+    void InstanceRegex(string content)
+    {
+        regex.IsMatch(content); // Noncompliant {{Use the 'RegexExtensions.SafeIsMatch' extension method.}}
+        regex.IsMatch(content, 0); // Noncompliant
+        regex.Matches(content); // Noncompliant {{Use the 'RegexExtensions.SafeMatches' extension method.}}
```

Comment:
The message generation seems unnecessary complicated to me, given that it gives you a static class and extension at the end, which is confusing.
Let's report just this in the extension case
```suggestion
        regex.Matches(content); // Noncompliant {{Use the 'SafeMatches' instead.}}
```
and this in the static case
```
Regex.IsMatch(content, "pattern"); // Noncompliant {{Use the 'SafeRegex.IsMatch' instead.}}
```
It will provide message that is more clear and will simplify the logic

Reply:
I will change the location depending on whether the method is static or not.